### PR TITLE
User story 4

### PR DESCRIPTION
**Pull request by nikokreissl**
- Create `ArtPieceDetails` component
- Create `[slug].js` where `ArtPieceDetails` component is rendered
- In `[slug].js` render art piece based on `slug` which is retrieved by `useRouter()`
- Add tests for `ArtPieceDetails` component

### CODE REVIEW

File: components/ArtPiecesDetails/ArtPieceDetails.test.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  expect(
+    screen.getByText("Orange Red and Green Abstract Painting")
+  ).toBeInTheDocument();
+  expect(screen.getByText("Steve Johnson")).toBeInTheDocument();
+  expect(screen.getByText("2018")).toBeInTheDocument();
+  expect(screen.getByText("Abstract Painting")).toBeInTheDocument();
+  expect(
+    screen.getByAltText("Orange Red and Green Abstract Painting")
+  ).toBeInTheDocument();
+  expect(screen.getByText("Back to Art Pieces")).toBeInTheDocument();
```
Comment (lenzloesche) [inline components/ArtPiecesDetails/ArtPieceDetails.test.js:24]: I am not sure if you should write a test for every expect.
also you didnt test if "back to art pieces" is a button.

File: components/ArtPiecesDetails/index.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+export default function ArtPiecesDetails({
+  image,
+  title,
+  artist,
+  year,
+  genre,
+}) {
+  return (
+    <main>
+      <Link href="/art-pieces">Back to Art Pieces</Link>
```
Comment (lenzloesche) [inline components/ArtPiecesDetails/index.js:13]: its not a button :)

File: pages/art-pieces/[slug].js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import { useRouter } from "next/router";
+import ArtPiecesDetails from "../../components/ArtPiecesDetails";
+
+import useSWR from "swr";
+
+export default function ArtPiecesDetailsPage() {
+  const router = useRouter();
+  const { slug } = router.query;
+  console.log(slug);
+  const { data: pieces } = useSWR("https://example-apis.vercel.app/api/art");
```
Comment (lenzloesche) [inline pages/art-pieces/[slug].js:11]: I am not sre, but the fetching logic should be in the app.js according to user story 3.

**Review by lenzloesche — COMMENTED** [review]
well done :)

File: components/ArtPiecesDetails/ArtPieceDetails.test.js (hunk 1 above)
Comment (nikokreissl) [reply components/ArtPiecesDetails/ArtPieceDetails.test.js:24]: We realised this too and added it 😊 Thanks for the hint!

File: pages/art-pieces/[slug].js (hunk 3 above)
Comment (nikokreissl) [reply pages/art-pieces/[slug].js:11]: You are right and we already discussed about it in the team and will update it with the next user story 😊

**Pull request merged**